### ISSUE TITLE
False positive: EmptyLineSeparator reports violation on method call

### ISSUE THREAD

**Issue by Praveen7294** [checkstyle/checkstyle#18438]
reported at https://github.com/checkstyle/checkstyle/pull/18311#issuecomment-3691084557

I have read check documentation: https://checkstyle.org/checks/whitespace/emptylineseparator.html#EmptyLineSeparator
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

`EmptyLineSeparator` reports violation on method call which wrong. It only occurs when there are more than one empty lines before a method call.

#### Config:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="EmptyLineSeparator">
      <property name="tokens"
               value="PACKAGE_DEF, IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF,
                    STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF, RECORD_DEF,
                    COMPACT_CTOR_DEF"/>
      <property name="allowNoEmptyLineBetweenFields" value="true"/>
      <property name="allowMultipleEmptyLines" value="false"/>
      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
    </module>
  </module>
</module>
```

#### Test.java:
```
import org.junit.Test;
import org.junit.Assert;

public class Test {
  
  public void foo(int a) {}

  @Test
  public void testFoo() {

    
    int a = 10;


    Test t = new Test();

    
    t.foo(10);
    int b = 20;


    Assert.assertFalse(false);
  }
}
```

#### Actual Cli:
```
$ java -jar checkstyle-12.3.0-all.jar -c config_check.xml Test.java
Starting audit...
[ERROR] F:\GitHub\headhtmltagname\Test.java:9:25: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:12:15: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:18:6: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:22:11: There is more than 1 empty line after this line. [EmptyLineSeparator]
Audit done.
Checkstyle ends with 4 errors.
```
According to the reported violation, lines 18 and 22 have more than one empty line after them, but there are no empty lines after these lines.

#### Expected Cli:
```
$ java -jar checkstyle-12.3.0-all.jar -c config_check.xml Test.java
Starting audit...
[ERROR] F:\GitHub\headhtmltagname\Test.java:9:25: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:12:15: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:15:6: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:19:11: There is more than 1 empty line after this line. [EmptyLineSeparator]
Audit done.
Checkstyle ends with 4 errors.
```
In my opinion, it might be incorrect line referencing.

**Comment by Zopsss** [comment]
@Praveen7294 can you investigate the reason behind these false-positives?

**Comment by Praveen7294** [comment]
> [@Praveen7294](https://github.com/Praveen7294) can you investigate the reason behind these false-positives?

I debugged, i found there is a problem in `getEmptyLine()` method inside `EmptyLineSeparatorCheck`. We need to fix that method.

**Comment by Zopsss** [comment]
If you have free time, can you please fix it?

**Comment by Praveen7294** [comment]
> If you have free time, can you please fix it?

Yes, i will try.

**Comment by Praveen7294** [comment]
@Zopsss @romani I have resolved this issue and am waiting for approval before submitting a PR.

**Comment by Zopsss** [comment]
@Praveen7294 sorry for the late reply, please point out the line numbers in the example. The CLI output is confusing without knowing the line numbers.

**Comment by Praveen7294** [comment]
@Zopsss 

**code:**
```
    Test t = new Test();                            // 15
                                                    // 16
                                                    // 17
    t.foo(10);                                      // 18
    int b = 20;                                     // 19
                                                    // 20
                                                    // 21
    Assert.assertFalse(false);                      // 22
  }                                                 // 23
}                                                   // 24
```

**Error**
```
[ERROR] F:\GitHub\headhtmltagname\Test.java:18:6: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:22:11: There is more than 1 empty line after this line. [EmptyLineSeparator]
```

**Expected Error**
```
[ERROR] F:\GitHub\headhtmltagname\Test.java:15:6: There is more than 1 empty line after this line. [EmptyLineSeparator]
[ERROR] F:\GitHub\headhtmltagname\Test.java:19:11: There is more than 1 empty line after this line. [EmptyLineSeparator]
```

don't go column number it is a random.

**Comment by Zopsss** [comment]
Thanks for the update. 

@romani I am approving this issue.

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/blob/990f30703694f134a77734a9c50d79c3a427bc97/src/main/resources/google_checks.xml#L203-L210

is this defect affecting goolgle style ? if yes, lets add few samples of Inputs for google's ITs, to make sure no side effects to google style.

**Comment by Praveen7294** [comment]
> is this defect affecting goolgle style ? if yes, lets add few samples of Inputs for google's ITs, to make sure no side effects to google style.

No, for reason see : https://github.com/checkstyle/checkstyle/pull/18545#issuecomment-3826126848
